bolts/data/datasets/audio/ecoacoustics.py

```diff
@@ -1 +1,224 @@
-# Placeholder for concrete custom dataset for the ecoacoustics project.
+"""Ecoacostics dataset provided A. Eldridge et al.
+    Alice Eldridge, Paola Moscoso, Patrice Guyot, & Mika Peck. (2018).
+    Data for "Sounding out Ecoacoustic Metrics: Avian species richness
+    is predicted by acoustic indices in temperate but not tropical
+    habitats" (Final) [Data set].
+    Zenodo. https://doi.org/10.5281/zenodo.1255218
+"""
+from __future__ import annotations
+import io
+from os import mkdir
+from pathlib import Path
+from typing import Callable, ClassVar, Optional, Union
+import zipfile
+
+from kit import parsable
+import pandas as pd
+import requests
+import torch
+from torch import Tensor
+import torchaudio
+import torchaudio.functional as F
+import torchaudio.transforms as T
+from typing_extensions import Literal
+
+from bolts.data.datasets.audio.base import PBAudioDataset
+from bolts.data.datasets.utils import AudioTform, FileInfo
+
+__all__ = ["Ecoacoustics"]
+SoundscapeAttr = Literal["habitat", "site"]
```

Comment:
If this is going to be kept for now, would you mind making it into an Enum? The `target_attr` inti argument can then be of type `Union[str, SoundscapeAttr]` with a casting from str to enum  done with `str_to_enum` in the init (see NICO/CelebA for an example)